starter-code/js/index.js

```diff
@@ -35,6 +61,17 @@ function createNewItemRow(itemName, itemUnitPrice){
 }
 
 function createNewItem(){
+//   var original = document.getElementById('repeat-this');
+//   var newItem = document.createElement(item);
+var i = 0;
+//   var clone = original.cloneNode(true);
+//   clone.id = "repeat-this" + ++i; 
+
+  // original.parentNode.appendChild(clone);
+var original = document.getElementById('repeat-this');
+var clone = original.cloneNode(true); // "deep" clone
```

Comment:
Tricky, but not best way to do it. 
If you clone a row that has a QTY assigned you will clone QTY's value too.
If there are no products in de cart you can't clone any product.

To improve the code doing it in this way, you can create a hidden row in the position 0 of the cart array, and you'll always copy this one :wink: